ethpm/contract.py

```diff
@@ -0,0 +1,128 @@
+from typing import Any, Dict, List  # noqa: F401
+
+import cytoolz
+from eth_utils import is_address, to_canonical_address
+from web3 import Web3
+from web3.contract import Contract
+
+from ethpm.exceptions import BytecodeLinkingError
+
+
+class LinkableContract(Contract):
+    """
+    A subclass of web3.contract.Contract that is capable of handling
+    contract factories with link references in their package's manifest.
+    """
+
+    deployment_link_refs = None
+    runtime_link_refs = None
+    is_bytecode_linked = False
+
+    def __init__(self, address: bytes = None, **kwargs: Dict[str, Any]) -> None:
+        super(LinkableContract, self).__init__(address=address, **kwargs)
+
```

Comment:
@carver I'm having some trouble with subclassing `LinkableContract` as both a factory and an instance.  I'd like to have the `LinkableContract` methods (ie `has_linkable_bytecode`/`link_bytecode`) available on the factory. 

Right now the way i'm doing this is passing in the factory as the `self` arg (i.e. `factory.has_linkable_bytecode(factory)`) but i'd much rather just call `factory.has_linkable_bytecode()` (which now raises a `missing 1 posarg self argument` error), any thoughts on how I could do this?

Reply:
Discussed in side channel. tl;dr - use `@combomethod`